jax/random.py

```diff
@@ -14,10 +14,30 @@
 
 """JAX pseudo-random number generators (PRNGs).
 
-The JAX PRNG system is based on "Parallel random numbers: as easy as 1, 2, 3"
-(Salmon et al. 2011). For details on the design and its motivation, see:
-
-https://github.com/google/jax/blob/master/design_notes/prng.md
+Usage:
+
+```python
+rng = jax.random.PRNGKey(seed)
+for i in xrange(num_steps):
+  rng, rng_input = jax.random.split(rng)
+  params = compiled_update(rng_input, params, next(batches))
+```
+
+Context:
+
+Among other requirements, Jax PRNG aims at:
+(a) ensure reproducibility,
+(b) parallelize well, both in terms of vectorization (generating array values)
+and multi-replica, multi-core computation. In particular it should not use 
+sequencing constraints between random function calls.
+
+The approach is based on:
+1. "Parallel random numbers: as easy as 1, 2, 3" (Salmon et al. 2011)
+2. "Splittable pseudorandom number generators using cryptographic hashing"
+(Claessen et al. 2013)
+
+See also https://github.com/google/jax/blob/master/design_notes/prng.md
+for the historical design and its motivation.
```

Comment:
Remove "historical" ?

Is it out of date in some way? If so, we should update it.

Reply:
I think the design no longer matches what has been implemented, e.g.

```
rng = lax.rng.new_rng()
for i in xrange(num_steps):
  rng, rng_input = lax.rng.split(rng)
```

So it's kind of explaining the design at some point in time, but does not match exactly what has been done.